File: code/ruby/print.rb
#!/usr/bin/env ruby
# coding: utf-8

require 'mongo'
require 'json'

Encoding.default_external = Encoding::UTF_8

def Main()

	client = Mongo::Client.new(['127.0.0.1:27017'], :database => 'test')
	collection = client['sakaguradb']
	collection.find.each do |e|
		# _id があると json としては違反
		# e.delete('_id')
		puts JSON.pretty_unparse(e, :indent=>'    ')
		# puts e['会社名']
	end

end

Main()

File: code/python/print.py
#!/usr/bin/env python
# coding: utf-8


import sys
import pymongo
import json
import codecs

#def Print(*args):
#	out = codecs.getwriter('utf-8')(sys.stdout)
#	for e in args:
#		out.write(e)
#	out.write("\n")

def Main():

	print '### 開始 ###'

	client = pymongo.MongoClient('localhost', 27017, )
	db = client['test']
	collection = db['sakaguradb']
	for e in collection.find():
		e['_id'] = "..."
		# del e['_id']
		print json.dumps(e, ensure_ascii=False, sort_keys=True, indent=4)

	print '--- end ---'

Main()

